src/hooks/useStateWithCallback.js

```diff
@@ -0,0 +1,26 @@
+import {useEffect, useRef, useCallback} from 'react';
+import _ from 'underscore';
+
+function useStateWithCallback(state, setState) {
+    const callbackRef = useRef(null);
+
+    const enhancedSetState = useCallback(
+        (value, callback) => {
+            callbackRef.current = callback;
+            setState(value);
+        },
+        [setState],
+    );
+
+    useEffect(() => {
+        if (!_.isFunction(callbackRef.current)) {
+            return;
+        }
+        callbackRef.current(state);
+        callbackRef.current = null;
+    }, [enhancedSetState, state]);
+
+    return enhancedSetState;
+}
```

Comment:
Yes, If we have a queue of `setStates` this `useEffect` will be called with the last state

Reply:
Should we place a comment about this in the code? 

This hook is very useful in some cases (like this), where we do not have such a scenario

Reply:
I think it can be more predictable and easier to control if I create a hook with a new state. After I did some additional checks, I think it will be better than wrapping the existing one.

What do you think?